packages/two_dimensional_scrollables/lib/src/tree_view/render_tree.dart

```diff
@@ -0,0 +1,744 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection' show LinkedHashMap;
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'tree_core.dart';
+import 'tree_delegate.dart';
+import 'tree_span.dart';
+
+// Used during paint to delineate animating portions of the tree.
+typedef _PaintSegment = ({int leadingIndex, int trailingIndex});
+
+/// A render object for viewing [RenderBox]es in a tree format that extends in
+/// both the horizontal and vertical dimensions.
+///
+/// [RenderTreeViewport] is the visual workhorse of the [TreeView]. It
+/// displays a subset of its [TreeViewNode] rows according to its own dimensions
+/// and the given [verticalOffset] and [horizontalOffset]. As the offset varies,
+/// different nodes are visible through the viewport.
+class RenderTreeViewport extends RenderTwoDimensionalViewport {
+  /// Creates a viewport for [RenderBox] objects in a tree format of rows.
+  RenderTreeViewport({
+    required Map<UniqueKey, TreeViewNodesAnimation> activeAnimations,
+    required Map<int, int> rowDepths,
+    required TreeViewTraversalOrder traversalOrder,
+    required double indentation,
+    required super.horizontalOffset,
+    required super.horizontalAxisDirection,
+    required super.verticalOffset,
+    required super.verticalAxisDirection,
+    required TreeRowDelegateMixin super.delegate,
+    required super.childManager,
+    super.cacheExtent,
+    super.clipBehavior,
+  })  : _activeAnimations = activeAnimations,
+        _rowDepths = rowDepths,
+        _traversalOrder = traversalOrder,
+        _indentation = indentation,
+        assert(verticalAxisDirection == AxisDirection.down &&
+            horizontalAxisDirection == AxisDirection.right),
+        super(
+          mainAxis: traversalOrder == TreeViewTraversalOrder.depthFirst
```

Comment:
Why does the main axis depend on traversalOrder?

Reply:
Because main axis is used in the super class to determine which order to visit children in.